File: examples/node/web_socket.js
var faye   = require('../../build/faye-node'),
    port   = process.argv[2] || 7000,
    socket = new faye.WebSocket.Client('socket://localhost:' + port + '/')

socket.onopen = function(event) {
  console.log('open')
  socket.send(JSON.stringify({channel: '/meta/subscribe', subscription: '/foo'}))
}

socket.onmessage = function(event) {
  console.log('message', event.data)
  socket.close(1002, 'Going away')
}

socket.onclose = function(event) {
  console.log('close', event.code, event.reason)
}

File: examples/ruby/web_socket.rb
require 'rubygems'
require File.expand_path('../../../lib/faye', __FILE__)

port = ARGV[0] || 7000

EM.run {
  socket = Faye::WebSocket::Client.new("socket://localhost:#{port}/")
  
  socket.onopen = lambda do |event|
    p [:open]
    socket.send JSON.dump('channel' => '/meta/handshake')
  end
  
  socket.onmessage = lambda do |event|
    p [:message, event.data]
    socket.close 1002, 'Going away'
  end
  
  socket.onclose = lambda do |event|
    p [:close, event.code, event.reason]
    EM.stop
  end
}
